lib/pbench/test/unit/server/test_user_auth.py

```diff
@@ -0,0 +1,365 @@
+                lastname="lastName",
+                email="user@domain.com",
+                password="12345",
+            )
+            data_register = resp_register.json
+
+            response = login_user(client, server_config, "username", "12345")
+            data_login = response.json
+            assert data_login["status"] == "success"
+            assert data_login["message"] == "Successfully logged in."
+            response = client.get(
+                f"{server_config.rest_uri}/user/username",
+                headers=dict(Authorization="Bearer " + data_login["auth_token"]),
+            )
+            data = response.json
+            assert data["status"] == "success"
+            assert data["data"] is not None
+            assert data["data"]["email"] == "user@domain.com"
+            assert response.status_code == 200
+
+    @staticmethod
+    def test_malformed_auth_token(client, server_config):
+        """ Test for user status for malformed auth token"""
+        with client:
+            resp_register = register_user(
+                client,
+                server_config,
+                username="username",
+                firstname="firstname",
+                lastname="lastName",
+                email="user@domain.com",
+                password="12345",
+            )
+            response = client.get(
+                f"{server_config.rest_uri}/user/username",
+                headers=dict(Authorization="Bearer" + "malformed"),
+            )
+            data = response.json
+            assert data is None
+
+    @staticmethod
+    def test_valid_logout(client, server_config):
+        """ Test for logout before token expires """
+        with client:
+            # user registration
+            resp_register = register_user(
+                client,
+                server_config,
+                username="user",
+                firstname="firstname",
+                lastname="lastName",
+                email="user@domain.com",
+                password="12345",
+            )
+            data_register = resp_register.json
+            assert data_register["status"] == "success"
+            assert data_register["auth_token"]
+            # user login
+            resp_login = login_user(client, server_config, "user", "12345")
+            data_login = resp_login.json
+            assert data_login["status"] == "success"
+            assert data_login["message"] == "Successfully logged in."
+            assert data_login["auth_token"]
+            # valid token logout
+            response = client.delete(
+                f"{server_config.rest_uri}/session",
+                headers=dict(Authorization="Bearer " + data_login["auth_token"]),
+            )
+            data = response.json
+            assert data["message"] == "Successfully logged out."
+            assert response.status_code == 200
+
+    # @staticmethod
+    # def test_invalid_logout(client, server_config):
```

Comment:
???

Also, "after token expires" is a good test case, but a regular logout -> logout sequence would also be useful.

In fact, another test case would be login -> login -> logout ... and show that the unlogged-out session is still good before showing that you can also log that one out and then be unauthorized. :-)

Reply:
yeah I commented it out for now, it need to be modified a little bit